scripts/zkai_attention_kv_d64_two_head_seq64_air_private_softmax_table_lookup_gate.py

```
@@ -0,0 +1,934 @@
+
+
+def fsync_parent_dir(path: pathlib.Path) -> None:
+    flags = getattr(os, "O_RDONLY", 0)
+    if hasattr(os, "O_DIRECTORY"):
+        flags |= os.O_DIRECTORY
+    try:
+        fd = os.open(path.parent, flags)
+    except OSError as err:
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(
+            f"failed to open artifact parent directory for fsync {path.parent}: {err}"
+        ) from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)